### PR TITLE
Update xgboost to 1.3.1

### LINKED ISSUE #467: Can't mix `featran-xgboost` dependency with newer versions of xgboost
```
java.lang.NoSuchMethodError: 'int ml.dmlc.xgboost4j.LabeledPoint.size()'
        at ml.dmlc.xgboost4j.java.DataBatch$BatchIterator.next(DataBatch.java:43)
        at ml.dmlc.xgboost4j.java.XGBoostJNI.XGDMatrixCreateFromDataIter(Native Method)
        at ml.dmlc.xgboost4j.java.DMatrix.<init>(DMatrix.java:54)
```

### PR DESCRIPTION
**Pull request by regadas**
Fixes #467

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -32,7 +32,7 @@ val scioVersion = "0.10.0-beta2"
 val simulacrumVersion = "1.0.1"
 val sparkVersion = "3.1.0"
 val tensorflowVersion = "1.15.0"
-val xgBoostVersion = "0.90"
+val xgBoostVersion = "1.3.1"
```
Comment (regadas) [inline build.sbt:35]: This is not really used anywhere and left it here just that we know which version are we targeting with the copied `LabeledPoint`

**Comment by regadas** [comment]
we will need to release 0.8.0 since this will not be been compat

**Comment by nevillelyh** [comment]
Is that why mima failed? Should we merge & fix that after?

**Comment by regadas** [comment]
@nevillelyh yea that's right! we can merge now

**Pull request merged**